src/main/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitter.kt

```diff
@@ -186,39 +190,34 @@ class HMPPSDomainEventsEmitter(
       .withAdditionalInformation("currentPrisonStatus", releaseReason.currentPrisonStatus?.name)
   }
 
-  private fun toCaseNotePublished(event: OffenderEvent): HmppsDomainEvent? {
+  private fun toCaseNotePublished(event: CaseNoteDeletedOffenderEvent): HmppsDomainEvent? {
     // SDI-594: If there is no offender id then this means that the case note has actually been removed instead
     // This means that we can ignore this event - will be handled by the offender deletion event instead.
-    if (event.offenderIdDisplay == null) {
-      log.warn(
-        "Ignoring case note published event for case note {} as offender id display is null",
-        event.caseNoteId,
-      )
-      return null
-    }
```

Comment:
I guess that this currently only happens for event type `OFFENDER_CASE_NOTES-DELETED`? Just seems an odd way to detect deletions instead of just checking the event type...

Reply:
I think I need to investigate further, does sound odd